### PR TITLE
Add triplanar node

### PR DESCRIPTION
**Pull request by luisbarrancos**
With randomization. Without connections to ID manifold defaults to using the assembly instance name to create a hash and ID to drive the randomization. This allows one to large scenes of i.e, procedurally modeled pebbles, stones, whatever, without a nice UV/ST parameterization, and have triPlanar deal with the texturing while at the same time avoiding having two identically shaded objects.

### CODE REVIEW

File: src/appleseed.shaders/include/appleseed/color/as_color_helpers.h [hunk 1]
```diff
@@ -314,4 +316,71 @@ void initialize_RGB_primaries(
...
+    else if (eotf == "Rec.2020")
+    {
+        scene_linear_cms = Rec2020_EOTF(input);
+    }
+    else
+    {
+#ifdef DEBUG
+        string shadername = "";
+        getattribute("shader:shadername", shadername);
+        warning("[WARNING!]: Unknown OETF mode %s, in %s, %s:%d\n",
```
Comment (dictoon) [inline src/appleseed.shaders/include/appleseed/color/as_color_helpers.h]: Remove exclamation mark, other warning messages don't have one.

File: src/appleseed.shaders/include/appleseed/texture/as_texture_helpers.h [hunk 2]
```diff
@@ -111,4 +111,43 @@ string get_interpolation_method(int method)
...
+
+    if (angle != 0.0)
+    {
+        rotate2d(st[0], st[1], angle, st[0], st[1]);
+    }
+
+    st[0] = sflip ? 1.0 - mod(st[0], UVWRAP) : mod(st[0], UVWRAP);
+    st[1] = tflip ? 1.0 - mod(st[1], UVWRAP) : mod(st[1], UVWRAP);
+
+    st[0] *= width + x_offset;
```
Comment (dictoon) [inline src/appleseed.shaders/include/appleseed/texture/as_texture_helpers.h]: Are you sure you want to multiply by `(width + x_offset)`?

File: src/appleseed.shaders/include/appleseed/transform/as_transform_helpers.h [hunk 3]
```diff
@@ -110,4 +110,20 @@ void compute_id_manifold(
...
+    float x,
+    float y,
+    float angle_in_degrees,
+    output float rx,
+    output float ry)
+{
+    float angle_rad = radians(angle_in_degrees);
+    float c, s;        
+    sincos(angle_rad, s, c);
+    float xx = x * c - s * y;
```
Comment (dictoon) [inline src/appleseed.shaders/include/appleseed/transform/as_transform_helpers.h]: Why not assign directly to `rx` and `ry`?

File: src/appleseed.shaders/src/appleseed/as_triplanar.osl [hunk 4]
```diff
@@ -0,0 +1,874 @@
...
+        string label = "Randomization",
+        string page = "Projection.Randomization"
+    ]],
+    int in_manifold = 0
+    [[
+        string as_maya_attribute_name = "manifold",
+        string as_maya_attribute_short_name = "man",
+        string label = "Manifold",
+        string page = "Projection.Randomization",
+        string help = "If a idManifold is connected, then it's used, otherwise the default assembly instance name ID is used instead."
```
Comment (dictoon) [inline src/appleseed.shaders/src/appleseed/as_triplanar.osl]: "name ID"? I can imagine *name* or *ID* but what "name ID" means?

Comment (luisbarrancos) [reply src/appleseed.shaders/src/appleseed/as_triplanar.osl]: Let me rephrase that. It's a hash built from object:assembly_instance_name.

File: src/appleseed.shaders/include/appleseed/texture/as_texture_helpers.h (hunk 2 above)
Comment (luisbarrancos) [reply src/appleseed.shaders/include/appleseed/texture/as_texture_helpers.h]: no, that is a bug, thanks :)
